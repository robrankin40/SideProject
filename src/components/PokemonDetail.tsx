import React from 'react';
import {Pokemon} from '../types';
import {View, Text, StyleSheet, Image} from 'react-native';

interface PokemonDetailProps {
  detail: Pokemon;
}

export const PokemonDetail: React.FC<PokemonDetailProps> = ({detail}) => {
  return (
    <View style={styles.wrapper}>
      <Text>{detail.name}</Text>
      <Text>Base Experience: {detail.base_experience}</Text>
      <Text>Weight: {detail.weight}</Text>
      <Text>Height: {detail.height}</Text>
      <Text>
        Abilities:{' '}
        {detail.abilities.map(({ability}) => ability.name).join(', ')}
      </Text>
      <Text>Forms: {detail.forms.map(({name}) => name).join(', ')}</Text>
      <Text>Moves: {detail.moves.map(({move}) => move.name).join(', ')}</Text>
      <Image
        source={{uri: detail.sprites.front_default}}
        style={styles.sprite}
      />
      <Image
        source={{uri: detail.sprites.back_default}}
        style={styles.sprite}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    margin: 16,
    padding: 16,
    borderRadius: 8,
    borderWidth: 1,
    borderColor: 'black',
  },
  sprite: {
    width: 128,
    height: 128,
  },
});
