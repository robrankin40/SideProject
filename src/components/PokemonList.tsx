import React from 'react';
import {FlatList, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import {APIResult} from '../types';

export interface PokemonListProps {
  data?: APIResult[];
  loading: boolean;
  hasError: boolean;
  onPressPokemon?: (pokemon: string) => void;
}

const EmptyState: React.FC<{loading: boolean; hasError: boolean}> = ({
  loading,
  hasError,
}) => (
  <View>
    {loading ? (
      <Text>Loading...</Text>
    ) : hasError ? (
      <Text>Unexpected Error</Text>
    ) : (
      <Text>No Pokemons</Text>
    )}
  </View>
);

export const PokemonList: React.FC<PokemonListProps> = ({
  data,
  loading,
  hasError,
  onPressPokemon,
}) => {
  const renderItem = ({item}: {item: APIResult; index: number}) => (
    <TouchableOpacity
      style={styles.itemWrapper}
      onPress={() => onPressPokemon?.(item.name)}
      testID={`button_${item.name}`}>
      <Text>{item.name}</Text>
    </TouchableOpacity>
  );
  return (
    <FlatList
      data={data}
      renderItem={renderItem}
      keyExtractor={(item, index) => `pokemon_${item.name}_${index}`}
      ListEmptyComponent={<EmptyState loading={loading} hasError={hasError} />}
    />
  );
};

const styles = StyleSheet.create({
  itemWrapper: {
    padding: 16,
    margin: 16,
    borderRadius: 8,
    borderWidth: 1,
    borderColor: 'black',
  },
});
