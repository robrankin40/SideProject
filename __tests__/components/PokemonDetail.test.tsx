import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {it, expect} from '@jest/globals';
import {mockPokemonDetail} from '../../mock';
import {PokemonDetail} from '../../src/components/PokemonDetail';

it('Pokemon Details Renders Correctly', () => {
  render(<PokemonDetail detail={mockPokemonDetail} />);
  expect(screen).toMatchSnapshot();
});
