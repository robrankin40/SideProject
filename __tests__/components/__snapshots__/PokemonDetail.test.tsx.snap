// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Details Renders Correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderRadius": 8,
      "borderWidth": 1,
      "margin": 16,
      "padding": 16,
    }
  }
>
  <Text>
    bulbasaur
  </Text>
  <Text>
    Base Experience: 
    64
  </Text>
  <Text>
    Weight: 
    69
  </Text>
  <Text>
    Height: 
    7
  </Text>
  <Text>
    Abilities:
     
    overgrow, chlorophyll
  </Text>
  <Text>
    Forms: 
    bulbasaur
  </Text>
  <Text>
    Moves: 
    razor-wind, swords-dance, cut, bind, vine-whip, headbutt, tackle, body-slam, take-down, double-edge, growl, strength, mega-drain, leech-seed, growth, razor-leaf, solar-beam, poison-powder, sleep-powder, petal-dance, string-shot, toxic, rage, mimic, double-team, defense-curl, light-screen, reflect, bide, sludge, skull-bash, amnesia, flash, rest, substitute, snore, curse, protect, sludge-bomb, mud-slap, outrage, giga-drain, endure, charm, false-swipe, swagger, fury-cutter, attract, sleep-talk, return, frustration, safeguard, sweet-scent, synthesis, hidden-power, sunny-day, rock-smash, facade, nature-power, helping-hand, ingrain, knock-off, secret-power, weather-ball, grass-whistle, bullet-seed, magical-leaf, natural-gift, worry-seed, seed-bomb, energy-ball, leaf-storm, power-whip, captivate, grass-knot, venoshock, round, echoed-voice, grass-pledge, work-up, grassy-terrain, confide, grassy-glide
  </Text>
  <Image
    source={
      {
        "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
      }
    }
    style={
      {
        "height": 128,
        "width": 128,
      }
    }
  />
  <Image
    source={
      {
        "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
      }
    }
    style={
      {
        "height": 128,
        "width": 128,
      }
    }
  />
</View>
`;
